File: language/gpl/java/jdk/java/lang/Runnable.java
// Interface

// Function: 0 input -> 0 output (void)

class Main {
  public static void main(String[] args) {

    Runnable r1 = () -> System.out.println("Hello, World!");
    Runnable r2 = new Runnable() {
      @Override
      public void run() {
        System.out.println("Hello, World!");
      }
    };

    r1.run();
    r2.run();
  }
}

File: machine-learning/llm/langchain/demo/packages/langchain_core/runnables/classes/Runnable.py
from langchain.chat_models import init_chat_model
from langchain_core.prompts import ChatPromptTemplate
from langchain_core.runnables import Runnable


def main():

    model = init_chat_model("google_genai:gemini-2.5-flash-lite")

    prompt_template = ChatPromptTemplate(
        [
            ("system", "You are a helpful AI bot. Your name is {name}."),
            ("human", "Hello, how are you doing?"),
            ("ai", "I'm doing well, thanks!"),
            ("human", "{user_input}"),
        ]
    )

    # Pipe together the prompt_template into the model and invoke
    chain: Runnable = prompt_template | model
    # chain: Runnable = prompt_template.__or__(model) # same

    response = chain.invoke(
        input={
            "name": "Bob",
            "user_input": "What is your name?",
        }
    )

    print(response.content)


if __name__ == "__main__":
    main()
